### PR TITLE
PR-3422

### PR DESCRIPTION
**Pull request by yasuflatland-lf**
@adolfopa According to the BChan's review, revised code.
- Got rid of private JSONUtil, mapping manuary
- Make the POJO fields protected
- Refactored according to DSCustomFieldManager and other *Manger implementations.
- Made validation API configurable.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+		catch (IOException ioException) {
+			_log.error(
+				"Failed to call supported language list." +
+					System.lineSeparator() + ioException.getLocalizedMessage());
+
+			return Collections.emptyList();
+		}
+	}
+
+	protected String getSupportedLanguage(
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java:74]: Can you make the protected methods private?

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java [hunk 2]
```diff
@@ -38,30 +39,30 @@
...
+		return JSONUtil.toList(
+			jsonObject.getJSONArray("translations"),
+			customFieldJSONObject -> new Translation() {
+				{
+					detectedSourceLanguage = customFieldJSONObject.getString(
+						"detected_source_language");
+					text = customFieldJSONObject.getString("text");
+				}
+			},
+			_log);
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java:62]: You should not use dynamic initializers unless strictly necessary. Please add a constructor to Translation that takes the detectedSourceLanguage and the text.

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java [hunk 3]
```diff
@@ -38,30 +39,30 @@
...
 	}
 
-	private String _fetch(
+	protected String post(
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java:65]: Don't use protected methods unless you need them for subclasses or unit tests. Always prefer private methods/fields.

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+		return JSONUtil.toList(
+			jsonArray,
+			customFieldJSONObject -> new SupportedLanguage() {
+				{
+					language = customFieldJSONObject.getString("language");
+					name = customFieldJSONObject.getString("name");
+					supportsFormality = customFieldJSONObject.getBoolean(
+						"supports_formality");
+				}
+			},
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java:125]: Don't use a dynamic init. block. Add a constructor to SupportedLanguage that takes the necessary parameters.

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/model/SupportedLanguage.java [hunk 5]
```diff
@@ -23,33 +23,33 @@ public class SupportedLanguage {
...
-		_supportsFormality = supportsFormality;
+		this.supportsFormality = supportsFormality;
 	}
 
-	private String _language;
-	private String _name;
-	private Boolean _supportsFormality;
+	protected String language;
+	protected String name;
+	protected Boolean supportsFormality;
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/model/SupportedLanguage.java:53]: Add a constructor and make the fields private. If the setters are no longer necessary, remove them.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java (hunk 2 above)
Comment (yasuflatland-lf) [reply modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java:62]: @adolfopa but the code BChan pointed out using [this way though](https://github.com/liferay/liferay-portal/blob/8b6306ce5ca9aa8ca896ba2223864897c5488b55/modules/apps/digital-signature/digital-signature-impl/src/main/java/com/liferay/digital/signature/internal/manager/DSCustomFieldManagerImpl.java#L42-L61)), how come?

Comment (adolfopa) [reply modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java:62]: That code is not using a dynamic init. block. You need to either use setters or a constructor.

Comment (yasuflatland-lf) [reply modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java:62]: @adolfopa Oh, yeah, it was the interface, got it.

**Comment by Hanlf** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
Hi @yasuflatland-lf , please take a look at the SF error, it should be related to this PR, thanks!

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/model/Translation.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+ * @author Yasuyuki Takeo
+ */
+public class Translation {
+
+	public Translation(String detectedSourceLanguage, String text) {
+		_detectedSourceLanguage = detectedSourceLanguage;
+		_text = text;
+	}
+
+	@JsonProperty("detected_source_language")
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/model/Translation.java:29]: Now that you're mapping things manually, shouldn't you remove these annotations?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by Hanlf** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
ci:test:relevant

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+		throws IOException, PortalException {
+
+		Http.Options options = new Http.Options();
+
+		options.setLocation(
+			URLBuilder.create(
+				url
+			).addParameter(
+				"auth_key", authKey
+			).build());
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/SupportedLanguageManager.java:85]: Doesn't this need to be sorted? It should be between `addPart` and `setMethod`.

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+		throws IOException, PortalException {
+
+		Http.Options options = new Http.Options();
+
+		options.setLocation(
+			URLBuilder.create(
+				url
+			).addParameter(
+				"auth_key", authKey
+			).build());
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/manager/TranslationManager.java:74]: Same as before; the methods should be sorted alphabetically.

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+			_log.error(
+				"DeepL does not support " + compareLanguage +
+					". Abort processing translation.");
+
+			return false;
+		}
+
+		return true;
+	}
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java:196]: The "verify" name is too ambiguous. The naming should reflect what it is verifying; in this case, something like "_isLanguageSupported" seems more appropriate.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+	private String _getLanguageCode(String languageId) {
+		List<String> list = Arrays.asList(
+			StringUtil.split(languageId, CharPool.UNDERLINE));
+
+		return StringUtil.toUpperCase(list.get(0));
+	}
+
+	private boolean _isLanguageSupported(
+		List<String> deepLLanguages, String compareLanguage) {
```
Comment (adolfopa) [inline modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java:158]: Can you rename `compareLanguage` to `languageCode`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Review by adolfopa — APPROVED** [review]
ci:forward

**Comment by adolfopa** [comment]
@yasuflatland-lf sorry, conflicts. Can you regenerate the lang files? After that you can send it directly to Brian -- mention this pull and my user so that he knows everything is all right. Thanks!

**Comment by yasuflatland-lf** [comment]
@adolfopa I ran `blade gw buildLang` but no files changed. What's the proper way of regenerating lang files?

**Comment by adolfopa** [comment]
@yasuflatland-lf Have you updated your branch? Maybe there are changes upstream that you haven't fetched yet.

**Comment by yasuflatland-lf** [comment]
@adolfopa Ok, do I need to rebase on top of the current HEAD?

**Comment by adolfopa** [comment]
@yasuflatland-lf The quickest way of fixing this is to remove the old buildLang commit, rebase against current master and committing a fresh buildLang again.
If you prefer it, push-force the changes here and I will forward it manually.

**Comment by yasuflatland-lf** [comment]
@adolfopa I'll give it a try, hw can I remove the old buildLang commit?

**Comment by adolfopa** [comment]
You can do it with an interactive rebase. See https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase.

**Comment by yasuflatland-lf** [comment]
@adolfopa Removed the old build lang and rebase on the master, then add the messages and ran buildLang again.

**Comment by adolfopa** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**